### PR TITLE
re-arrange metacache struct to be optimal

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
re-arrange metacache struct to be optimal 

## Motivation and Context
re-arrange structs for less alignment size

## How to test this PR?
Nothing should be needed tests cover it.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/metacache.go [hunk 1]
```diff
@@ -54,20 +54,21 @@ const (
 
 //go:generate msgp -file $GOFILE -unexported
 
+//msgp:tuple metacache
```
Comment (klauspost) [inline cmd/metacache.go]: This breaks existing and makes upgrades hard to do. No real need for this.

File: cmd/metacache-stream.go [hunk 2]
```diff
@@ -56,14 +56,16 @@ const metacacheStreamVersion = 2
 
 // metacacheWriter provides a serializer of metacache objects.
 type metacacheWriter struct {
+	// do not re-arrange the struct
+	// this struct has been re-ordered
+	// to use less space.
```
Comment (klauspost) [inline cmd/metacache-stream.go]: No need for comment. This is not a critical struct.

**Review by klauspost — COMMENTED** [review]
Let's keep these fixes for structs that have a chance of having 1000+ allocated.

Rearranging `metacache` is fine, but don't change it to use tuples.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**